### PR TITLE
DS v3 - Migration Guide

### PR DESCRIPTION
**Pull request by EdwinGuzman**
## This PR does the following:

- Migration guide draft.
- https://nypl-design-system-git-react18-migration-guide-nypl.vercel.app/?path=%2Fdocs%2Fdevelopment-guide-v3-migration-guide--docs
- 
## Accessibility concerns or updates

-

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import { Meta, Source } from "@storybook/blocks";
+
+import Button from "../Button/Button";
+import Link from "../Link/Link";
+import Text from "../Text/Text";
+
+<Meta title="Development Guide/v3 Migration Guide" />
+
+# Design System v3 Migration Guide
```
Comment (bigfishdesign13) [inline src/components/DevelopmentGuide/V3MigrationGuide.mdx]: Change to `Reservoir v3 Migration Guide`.

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+### Main Additions and Upgrades
+
+- The `Menu` (new) and `MultiSelect` (v2) components are included in this release.
+- Chakra 1.8 does not support React 18, so Chakra has been updated to version
+  2.8. **IMPORTANT: If you manually and separately install Chakra in your app,
+  make sure to also upgrade it in the `package.json` file.**
+
+General updates can be found in the [Changelog](https://github.com/NYPL/nypl-design-system/blob/development/CHANGELOG.md)
+but main points to note:
```
Comment (bigfishdesign13) [inline src/components/DevelopmentGuide/V3MigrationGuide.mdx]: Add a comma and some other words...
```
General updates can be found in the [Changelog](https://github.com/NYPL/nypl-design-system/blob/development/CHANGELOG.md), but the main points to note are:
```

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  horizontal scrolling for mobile viewports.
+
+## Breaking Changes
+
+### Design System
+
+- The `NewsletterSignup` and `SocialMediaLinks` components have been removed.
+  They will be added back in a later version.
+- `SimpleGrid` gap values have been updated to include `"grid"` in the prop
+  value name. For example, `gap=”grid.xs”`.
```
Comment (bigfishdesign13) [inline src/components/DevelopmentGuide/V3MigrationGuide.mdx:94]: I'm still not sure what to do about this.  We do use the `grid.*` values internally in a few components, but I am not sure why we decided to use this pattern.  The `GridGaps` type is also used for spacing in the `Form` component.  Was there some weird conflict?

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+// The scrollIntoView function is necessary for the Tabs component.
+window.HTMLElement.prototype.scrollIntoView = jest.fn()
+
+// Related to the Slider component
+class ResizeObserver {
+observe() {}
+unobserve() {}
+disconnect() {
+}
```
Comment (bigfishdesign13) [inline src/components/DevelopmentGuide/V3MigrationGuide.mdx]: Nit: add two spaces to the lines of code within the class brackets.

File: src/theme/components/multiSelectItemsCountButton.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ const MultiSelectItemsCountButton = defineStyleConfig({
     top: { base: "12px", md: "10px" },
     width: "46px",
     zIndex: 10000,
-    py: 'xxxs',
+    py: "xxxs",
```
Comment (bigfishdesign13) [inline src/theme/components/multiSelectItemsCountButton.ts:21]: 👀

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Just needs some minor adjustments.

File: src/theme/components/multiSelectItemsCountButton.ts (hunk 5 above)
Comment (EdwinGuzman) [reply src/theme/components/multiSelectItemsCountButton.ts:21]: Hmm not sure how this happened.

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx (hunk 3 above)
Comment (EdwinGuzman) [reply src/components/DevelopmentGuide/V3MigrationGuide.mdx:94]: I just encountered this in Research Catalog with v2.1.6 actually. It's odd this did not occurred before. Here are more concrete examples using the v3 `rc`:

- https://github.com/NYPL/research-catalog/pull/115/files#diff-456863484b0393d53a579ae78a6e99086e7daa6fd0afdcc0c50e8fec2a0471eeR49
- https://github.com/NYPL/nypl-ds-test-app/pull/94/files#diff-a333ac33ea441affec39a9410d57a5d8c0e5d07e8e8bf3f1a4f9acb5900a0be7R59

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx (hunk 4 above)
Comment (EdwinGuzman) [reply src/components/DevelopmentGuide/V3MigrationGuide.mdx]: I tried but it doesn't work! I may need to break this up.

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx (hunk 3 above)
Comment (bigfishdesign13) [reply src/components/DevelopmentGuide/V3MigrationGuide.mdx:94]: After moving to DS v3, did those apps start throwing errors due to the prop value?  Is that why they added the `grid` prefix?

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+# Reservoir v3 Migration Guide
+
+## Table of Contents
+
+- {<Link href="#general-updates-requirements">General Updates & Requirements</Link>}
+- {<Link href="#upgrade-path">Upgrade Path</Link>}
+- {<Link href="#breaking-changes">Breaking Changes</Link>}
+- {<Link href="#additional-testing-needs">Additional Testing Needs</Link>}
+
+## General Updates & Requirements
```
Comment (oliviawongnyc) [inline src/components/DevelopmentGuide/V3MigrationGuide.mdx:18]: What do you think of actually including the command

```
$ npm install @nypl/design-system-react-components@latest
```

**Review by oliviawongnyc — APPROVED** [review]
Looks good! Will we be updating the `Welcome` page too? In the vein of my comment below, I also think it might be useful to link the repo's README in the `Welcome` page and add it more prominently. It seems like it should be really easy to figure out how to get started (either for the first time, or to update).

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx (hunk 3 above)
Comment (EdwinGuzman) [reply src/components/DevelopmentGuide/V3MigrationGuide.mdx:94]: The Research Catalog threw this error in 2.1.6, so not necessarily a v3 issue.

File: src/components/DevelopmentGuide/V3MigrationGuide.mdx (hunk 6 above)
Comment (EdwinGuzman) [reply src/components/DevelopmentGuide/V3MigrationGuide.mdx:18]: Oh that's a good point. Hmm I think that command will make it include the `^` in package.json which can lead to issues. I'll be explicit about it in this next update.

Comment (EdwinGuzman) [reply src/components/DevelopmentGuide/V3MigrationGuide.mdx:18]: Updated in "npm Package Update" section. Let me know what you think.

Comment (oliviawongnyc) [reply src/components/DevelopmentGuide/V3MigrationGuide.mdx:18]: Love it!

**Pull request merged**